tutorials/shaders/shader_reference/spatial_shader.rst

```diff
@@ -225,6 +238,20 @@ The default use of a Godot fragment processor function is to set up the material
 and to let the built-in renderer handle the final shading. However, you are not required to use all
 these properties, and if you don't write to them, Godot will optimize away the corresponding functionality.
 
+Below are examples of frequently used matrix inputs for shaders.
+
+.. code-block:: glsl
+
+    MODEL_MATRIX[3].xyz = Object's world space position // global_transform.origin in GDScript
+    // Using mat3(MODEL_MATRIX) is the same as the global_transform.basis in GDScript
+    INV_VIEW_MATRIX[3].xyz = Camera's world space position // camera.global_transform.origin in GDScript
+    VIEW_MATRIX[3].xyz = Camera eye vector in world space direction of the camera
```

Comment:
I wonder if this makes more sense in a table or some other format. Something like:

Many helpful values can be found within the Matrix built-ins themselves, here are a few:

| Built In                                              | Description                                                                                      |
|-------------------------------------------------------|--------------------------------------------------------------------------------------------------|
| ``MODEL_MATRIX[3].xyz``                               | Object's world space position. It is also equivalent to ``global_transform.origin`` in GDScript. |
| ``mat3(MODEL_MATRIX)``                                | etc.                                                                                             |
| ``INV_VIEW_MATRIX[3].xyz``                            | etc.                                                                                             |
| ``VIEW_MATRIX[3].xyz``                                | etc.                                                                                             |
| ``(MODELVIEW_MATRIX * vec4(0.0, 0.0, 0.0, 1.0)).xyz`` | etc.                                                                                             |

Reply:
Fair enough, I don't have a strong opinion one way or the other. I don't love the blend between code and text. So perhaps, it would make sense to do more psuedocode with a comment like in the shading reference.

e.g.: 
```
vec3 world_space_position = MODEL_MATRIX[3].xyz; // Object's world space position. This is also equivalent to global_transform.origin in GDScript.
```